server/routes/pt.js

```diff
@@ -114,4 +116,184 @@ router.post('/search', verifyToken, async (req, res) => {
   }
 });
 
+router.post('/addptrequest', verifyToken, async (req, res) => {
+  try {
+    const pt_id = req.body.pt_id; // ID of PT that was requested
+    const client_id = req.user.userId; // ID of client that made request
+ 
+    Pt.findByIdAndUpdate(pt_id,
+      {$push: {requests: client_id}},
+      {safe: true, upsert: true},
+      function(err,doc) {
+        if(err){
+          console.log(err);
+        }else{
+          return res.status(OK).send(client_id);
+        }
+      });
+  }
+  catch (err) {
+    return res.status(BAD_REQUEST).send(err.message);
+  }
+  
+});
+
+router.post('/addclientrequest', verifyToken, async (req, res) => {
+  try {
+    const pt_id = req.body.pt_id; // ID of PT that was requested
+    const client_id = req.user.userId; // ID of client that made request
+    
+    Client.findByIdAndUpdate(client_id,
+      {$push: {requests: pt_id}},
+      {safe: true, upsert: true},
+      function(err,doc) {
+        if(err){
+          console.log(err);
+        }else{
+          return res.status(OK).send(pt_id);
+        }
+      });
+  }
+  catch (err) {
+    return res.status(BAD_REQUEST).send(err.message);
+  }
+  
+});
+
+router.post('/getrequests', verifyToken, async (req, res) => {
+  try {
+    const id = req.user.userId;
+    const doc = await Pt.findById(id);
+    if (!doc) return res.status(NOT_FOUND).send();
+
+    const requests = doc.requests;
+    const names = [];
+    for await (const docs of User.findById(requests)) {
```

Comment:
```suggestion
    for await (const user of User.find({ _id: { $in: requests } })) {
      try {
        names.push(user.publicName);
      }
```
This replaces lines 171–175.